packages/react-components/src/components/table-output-component.tsx

```diff
@@ -653,4 +654,56 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
             });
         }
     }
+
+    private toggleColumnVisibility(columnApi: ColumnApi, column: ColDef) {
+        if (!columnApi || !column.field) {
+            return;
+        }
+
+        columnApi.setColumnsVisible([column.field], !columnApi.getColumn(column).isVisible());
+        const allCols = cloneDeep(this.state.tableColumns);
+
+        allCols.map( item => {
+            if (item.field === column.field) {
+                item.hide = columnApi.getColumn(column).isVisible();
+            }
+        });
+
+        this.setState({
+            tableColumns: allCols
+        });
+    }
+
+    protected showOptions(): React.ReactNode {
+        /* eslint-disable @typescript-eslint/no-non-null-assertion */
+        if (!this.columnApi) {
+            return;
+        }
+        const columns = this.columnApi.getAllColumns();
+
+        return <React.Fragment>
+            <table style={{width: '100%'}}>
+                <thead>
+                    <tr style={{textAlign: 'left'}}>
+                        <th>Show/Hide</th>
+                        <th>Column Name</th>
+                    </tr>
+                </thead>
```

Comment:
In my opinion, this table header is unnecessary and I would prefer the checkboxes much closer to the column names.